src/basic/unit-file.c

```diff
@@ -87,8 +87,8 @@ int unit_validate_alias_symlink_or_warn(int log_level, const char *filename, con
          * -ELOOP for an alias to self.
          */
 
-        src = basename(filename);
-        dst = basename(target);
+        path_extract_filename(filename, (char**)&src);
+        path_extract_filename(target, (char**)&dst);
```

Comment:
buffers for both are leaked, they need to be annotated with _cleanup_free_

Reply:
> buffers for both are leaked, they need to be annotated with _cleanup_free_

When I say "Thank you sir", I mostly always really do mean it. Very well.

One could spend hours revelling in confusion, or one could just be guided right by the experts.

Thank you so much sir.

Reply:
I was wondering initially why i had to use _clean_up_free for src and dst since they didn't have that attribute originally, then i decided to look at path_extract_filename() again, and I realized it calls strndup, and ultimately assigns the returned pointers to src and dst when they're passed to it. I hanven't used strndup before, but I've used strdup and i remember clearly that it calls malloc. 

Thank you so much for all of your guidance sir. Thank you.